### ISSUE TITLE
acc never change after every batch of learning(20 newsgroup dataset)

### ISSUE THREAD

**Issue by ronzillia** [fchollet/keras#4994]
I want to see the result for LSTM on the 20 Newsgroup data.
Everything is the same as [https://github.com/fchollet/keras/blob/master/examples/pretrained_word_embeddings.py](url)
I just replace the Conv part by LSTM
`

model = Sequential() <br />model.add(embedding_layer)
model.add(LSTM(128, dropout_W=0.2, dropout_U=0.2))  
model.add(Dense(1))
model.add(Activation('sigmoid'))
model.add(Dense(len(labels_index), activation='softmax'))
model.compile(loss='binary_crossentropy',
              optimizer='adam',
              metrics=['accuracy'])

`
Unfortunately, for every batch processed, I got the same acc as follow:
Epoch 1/1
 4384/15998 [=======>......................] - ETA: 1953s - loss: 0.1985 - acc: 0.9500
Epoch 1/1
 4480/15998 [=======>......................] - ETA: 1937s - loss: 0.1985 - acc: 0.9500
Epoch 1/1
 4544/15998 [=======>......................] - ETA: 1928s - loss: 0.1985 - acc: 0.9500

the acc is always 0.95.
I have no idea where went wrong.
Thank you all

**Comment by blucena** [comment]
The problem is that you are using a "binary_crossentropy" loss yet you are trying to do a multiclass classification.  Also, having the output dimension of your first Dense layer as 1 is probably not a good idea.  Try changing `model.add(Dense(1))` to `model.add(Dense(20))` (or 50) and changing `model.compile(loss='binary_crossentropy', ` to `model.compile(loss='categorical_crossentropy,`

**Comment by ronzillia** [comment]
@blucena  Thanks so much!
